### PR TITLE
Make ChibiOS lwIP platform independent

### PR DESCRIPTION
**Pull request by networkfusion**
## Description

Unlocks the ability to update LWIP independent of ChibiOS.
This PR uses the same release tag as the current version provided from ChibiOS to not break anything.
It notes that NXP targets use a very outdated version but would only (possibily) affect builds in a dev container (all) image.

## Motivation and Context
Currently LWIP on ChibiOS is locked to a specific version which has multiple bugs.
After comparing the current compressed lwip folder contained in `ext` against the same release version from its official source, they match exactly.

## Screenshots
image

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dev Containers (changes related with Dev Containers, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).

### CODE REVIEW

File: targets/FreeRTOS/NXP/CMakeLists.txt [hunk 1]
```diff
@@ -95,11 +95,11 @@ endif()
 
 if(NO_LWIP_SOURCE)
 
-    message(STATUS "lwIP ${LWIP_GIT_TAG} from savannah.nongnu.org repo")
+    message(STATUS "lwIP ${LWIP_GIT_TAG} from savannah.gnu.org repo")
```
Comment (Ellerbach) [inline targets/FreeRTOS/NXP/CMakeLists.txt:98]: why changing the URL?

**Review by Ellerbach — COMMENTED** [review]
All up looks good. Any version to change somewhere for the container? ;-)

File: targets/FreeRTOS/NXP/CMakeLists.txt (hunk 1 above)
Comment (networkfusion) [reply targets/FreeRTOS/NXP/CMakeLists.txt:98]: I was having issues with using the "nongnu" link. The funny thing is, if you browse to https://git.savannah.nongnu.org/cgit/lwip.git and see the link at the bottom, it is actually the one I have changed it to:

image

Also, there is an official mirror on GH: https://github.com/lwip-tcpip/lwip so we could adjust to that.

**Comment by networkfusion** [comment]
> All up looks good. Any version to change somewhere for the container? ;-)

Given it only changes the build system, was going to wait on that, or at least until the change is approved.

File: targets/FreeRTOS/NXP/CMakeLists.txt (hunk 1 above)
Comment (Ellerbach) [reply targets/FreeRTOS/NXP/CMakeLists.txt:98]: Found as well the mirror version on github. Been looking at it for mDNS ;-) As we're not contributing and only consumming, maybe better to use the GitHub mirror as it seems to work all the time compare to the others?

Comment (networkfusion) [reply targets/FreeRTOS/NXP/CMakeLists.txt:98]: Switched.

**Comment by josesimoes** [comment]
@networkfusion will you address my review comments?

**Comment by networkfusion** [comment]
> @networkfusion will you address my review comments?

I am not seeing any comments. Have you submitted the review?

File: .devcontainer/sources/Dockerfile.ChibiOS [hunk 2]
```diff
@@ -45,21 +45,24 @@ RUN apt-get update \
     srecord
 
 # Create needed directories
-RUN mkdir -p /usr/local/bin/gcc \
-    && mkdir -p /usr/local/bin/xtensa \
-    && mkdir -p /usr/local/bin/titools
+RUN mkdir -p /usr/local/bin/gcc
 
-# Clone repos for STM32 including AzureRTOS
+# Clone ChibiOS SVN
```
[the remark this answers is not in the document]
Comment (networkfusion) [reply .devcontainer/sources/Dockerfile.ChibiOS:52]: Reverted.

File: config/user-tools-repos.TEMPLATE.json [hunk 3]
```diff
@@ -13,14 +13,17 @@
...
                 "MBEDTLS_SOURCE": "<path-to-mbedtls-source-mind-the-forward-slashes>",
                 "FATFS_SOURCE": "<path-to-local-FATFS-source-mind-the-forward-slash>",
                 "SPIFFS_SOURCE": "<path-to-local-SPIFFS-source-mind-the-forward-slash>",
                 "ESP32_IDF_PATH": "<absolute-path-to-the-IDF-folder-mind-the-forward-slashes>",
                 "TI_SL_CC32xx_SDK_SOURCE": "<path-to-local-TI_SimpleLink-CC32xx-SDK-source-mind-the-forward-slash>",
                 "TI_SL_CC13xx_26xx_SDK_SOURCE": "<path-to-local-TI_SimpleLink-CC13xx_26xx-SDK-source-mind-the-forward-slash>",
                 "TI_XDCTOOLS_SOURCE": "<path-to-local-TI-XDC-Tools-source-mind-the-forward-slash>",
-                "TI_SYSCONFIG_SOURCE": "<path-to-local-TI-SYSCONFIG-Tool-source-mind-the-forward-slash>"
+                "TI_SYSCONFIG_SOURCE": "<path-to-local-TI-SYSCONFIG-Tool-source-mind-the-forward-slash>",
+                "AZURERTOS_SOURCE_FOLDER": "<path-to-local-AzureRTOS-source-mind-the-forward-slash>",
```
[the remark this answers is not in the document]
Comment (networkfusion) [reply config/user-tools-repos.TEMPLATE.json]: Reverted.

File: targets/ChibiOS/CMakeLists.txt [hunk 4]
```diff
@@ -209,6 +209,54 @@ if(NF_FEATURE_HAS_SDCARD OR NF_FEATURE_HAS_USB_MSD)
 
 endif()
 
+# if LWIP is enabled add it to the build
```
[the remark this answers is not in the document]
Comment (networkfusion) [reply targets/ChibiOS/CMakeLists.txt]: Fixed.

**Comment by networkfusion** [comment]
@josesimoes Hopefully you are happy now.

**Review by Ellerbach — APPROVED** [review]
Looks good to me!

**Comment by josesimoes** [comment]
Just build the F769 DISCO from this PR.

- Boots fine.
- According to the LEDs in the RJ45 connector, there is network activity.
- Running the TLS sample, it doesn't get past the wait for IP...

- Checking the device has network capabilities:
image

- Strangely the device name is wrongly reported in VS:
image

This requires further investigation. 
I suspect that some things changed in lwIP since the tested version, in what concerns the platform specifics and interface, causing this.

**Comment by networkfusion** [comment]
@josesimoes Supprised at that, since this has no changes when compared to chibiOS lwIP. I will run the same test on my 69I.

**Review by Ellerbach — APPROVED** [review]
Thanks, looks good to me!

**Comment by Ellerbach** [comment]
only thing missing is bimping the containers!

**Review by josesimoes — APPROVED** [review]
Just merged from upstream and made a new local build from a clean build folder.
Tested with TLS sample connecting to [https://g2cert.azure-devices.net](https://g2cert.azure-devices.net/) and all went fine.
Also did it with a couple more websites and everything seems to be working.

Maybe it was in a bad mood the other day... so... approving and merging this. 😉

**Comment by networkfusion** [comment]
@josesimoes  I have just noticed a merge conflict issue with the dev container using the old version of fatFS. I am going to correct it.

**Comment by networkfusion** [comment]
Done.

**Comment by josesimoes** [comment]
I've already fixed the merge conflicts. 😉

**Pull request merged**